### PR TITLE
Feature/mw 23

### PR DESCRIPTION
**Pull request by Aaliyah379**
Please review

### CODE REVIEW

File: MyWorkout.Web/ApiControllers/ExercisesApiController.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        }
+
+        // GET: api/ExercisesApi
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Exercise>>> GetExercises()
+        {
+            var exercise = await _unitOfWork.ExerciseRepository
+                .ReadAll()
+                .ToListAsync();
+            exercise.ForEach(x => _unitOfWork.Load(x, x => (IEnumerable<Exercise>)x.Repeats));
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/ExercisesApiController.cs]: `IEnumerable<Repeat>`

File: MyWorkout.Web/ApiControllers/ExercisesApiController.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        // DELETE: api/ExercisesApi/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<Exercise>> DeleteExercise(int id)
+        {
+            var exercise = await _unitOfWork.ExerciseRepository.Read(id);
+            if (exercise == null)
+            {
+                return NotFound();
+            }
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/ExercisesApiController.cs:102]: await ExerciseExists(id)

File: MyWorkout.Web/ApiControllers/PlansApiController.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        // GET: api/PlansApi
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Plan>>> GetPlans()
+        {
+            var plan = await _unitOfWork.PlanRepository
+               .ReadAll()
+               .ToListAsync();
+            plan.ForEach(x => _unitOfWork.Load(x, x => (IEnumerable<Plan>)x.WorkoutDays));
+
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/PlansApiController.cs:33]: `IEnumerable<WorkoutDay>`

File: MyWorkout.Web/ApiControllers/PlansApiController.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        // DELETE: api/PlansApi/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<Plan>> DeletePlan(int id)
+        {
+            var plan = await _unitOfWork.PlanRepository.Read(id);
+            if (plan  == null)
+            {
+                return NotFound();
+            }
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/PlansApiController.cs]: `await PlanExists(id);`

File: MyWorkout.Web/ApiControllers/PlansApiController.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+            await _unitOfWork.PlanRepository.Delete(id);
+            await _unitOfWork.Save();
+
+            return plan;
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/PlansApiController.cs]: return NoContent();

File: MyWorkout.Web/ApiControllers/RepeatsApiController.cs [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            var repeat = await _unitOfWork.RepeatRepository.Read(id);
+            if (repeat == null)
+            {
+                return NotFound();
+            }
+
+            await _unitOfWork.RepeatRepository.Delete(id);
+            await _unitOfWork.Save();
+
+            return repeat;
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/RepeatsApiController.cs]: return NoContent();

File: MyWorkout.Web/ApiControllers/RepeatsApiController.cs [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+           await _unitOfWork.Save();
+
+           return CreatedAtAction("GetRepeat", new { id = repeat.Id }, repeat);
+        }
+
+        // DELETE: api/RepeatsApi/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<Repeat>> DeleteRepeat(int id)
+        {
+            var repeat = await _unitOfWork.RepeatRepository.Read(id);
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/RepeatsApiController.cs]: await RepeatExists(id)

File: MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+            return CreatedAtAction("GetWorkoutDay", new { id = workoutDay.Id }, workoutDay);
+        }
+
+        // DELETE: api/WorkoutDaysApi/5
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<WorkoutDay>> DeleteWorkoutDay(int id)
+        {
+            var workoutDay = await _unitOfWork.WorkoutDayRepository.Read(id);
+            if (workoutDay == null)
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs]: await WorkoutDayExists(id)

File: MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+            {
+                return NotFound();
+            }
+
+            await _unitOfWork.WorkoutDayRepository.Delete(id);
+            await _unitOfWork.Save();
+
+            return workoutDay;
```
Comment (YuriyMorozyuk95) [inline MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs]: return NoContent();

**Review by YuriyMorozyuk95 — APPROVED** [review]

File: MyWorkout.Web/ApiControllers/ExercisesApiController.cs (hunk 1 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/ExercisesApiController.cs]: Updated

File: MyWorkout.Web/ApiControllers/PlansApiController.cs (hunk 3 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/PlansApiController.cs:33]: Updated

File: MyWorkout.Web/ApiControllers/PlansApiController.cs (hunk 4 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/PlansApiController.cs]: Updated

File: MyWorkout.Web/ApiControllers/PlansApiController.cs (hunk 5 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/PlansApiController.cs]: Updated

File: MyWorkout.Web/ApiControllers/RepeatsApiController.cs (hunk 6 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/RepeatsApiController.cs]: Updated

File: MyWorkout.Web/ApiControllers/RepeatsApiController.cs (hunk 7 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/RepeatsApiController.cs]: Updated

File: MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs (hunk 8 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs]: Updated

File: MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs (hunk 9 above)
Comment (Aaliyah379) [reply MyWorkout.Web/ApiControllers/WorkoutDaysApiController.cs]: Updated

**Pull request merged**